applications/parami-protocol.md

```diff
@@ -146,29 +146,65 @@ Yanping YANG
+| 0e. | Article | We will publish an article or medium that explains what was done as part of the grant). |
+| 1. | Pallet(s)/did | Build Parami ParaChain based on Substrate3.0.0, the core modules that have been developed include DID(Decentralized Identity) pallet, Decentralised identity is an emerging concept that gives back control of identity to consumers. |
+| 2. | Pallet(s)/bridge | Completed the bridge development between ETH ERC20 and Parami chain. Please refer to https://github.com/parami-protocol/parami/tree/main/pallets/bridge  |
+| 3. | Pallet(s)/nft | Complete NFT pallets, please refer to https://github.com/parami-protocol/parami/tree/main/pallets/nft |
+| 4. | Testnet | Setup and launch Parami testnet, which has been running stably for about a month. Parami testnet has has become a parachain to Rococo. Please refer to (https://polkadot.js.org/apps/?rpc=wss://rococo-rpc.polkadot. io#/parachains/parathreads). |
+
+### Milestone  2
+* **Estimated duration:** 2 month
+* **FTE:** 3
+* **Costs:** 0.33 BTC
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | We will provide inline documentation of the code and API documentation to help developers to use the wallet |
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 70%) to ensure functionality and robustness. In the guide we will describe how to run these tests |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality for wallet. |
+| 0e. | Article | We will publish an medium that explains what was done as part of the grant). |
+| 1. | Pallet(s)/offchain | Develop Offchain-worker to verify ads and incentives, rewards and withdrawals, etc.
+| 2. | Pallet(s)/socialcoin | Support Social coin and introduce NFT support for advertisers.
+| 3. | Pallet(s)/oracle | Refine oracle implementation. Engage ad-specific validation into it, like an anti-spam check, visibility validation, and so on. |
+| 4. | Parami Wallet | We will provide an available wallet products for users and developers| 
+| 5. | Explorer |  Build Parami Chain explorer,  and provide block and transaction query support;  |
+
+### Milestone  3
+* **Estimated duration:** 1 month
+* **FTE:** 3
+* **Costs:** 0.34 BTC 
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | We will provide inline documentation of the code and API documentation |
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 80%) to ensure functionality and robustness. In the guide we will describe how to run these tests |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality. |
+| 0e. | Article | We will publish an achieved that explains what was done). |
+| 1. | Pallet(s)/onchain | Develop on-chain governance logic to build a more decentralized platform. |
```

Comment:
Governance modules already exist. We won't fund "governance logic to build a more decentralized platform".

Reply:
hi @alxs ,  We spent quite a lot time exploring Zero Knowledge Proof and its adoption to advertising at the beginning of development, including studying cryptography with researchers in academia, which extended the duration. And we also finished some milestone in the last application.